ui/app/src/components/shared/notifications/NotificationItem.tsx

```diff
@@ -1,27 +1,78 @@
-import React, { useState } from 'react';
-import { Icon, ProgressIndicator, Link as FluentLink, Stack, DefaultPalette } from '@fluentui/react';
+import React, { useContext, useEffect, useState } from 'react';
+import { Icon, ProgressIndicator, Link as FluentLink, Stack, DefaultPalette, Shimmer, ShimmerElementType, MessageBar, MessageBarType } from '@fluentui/react';
 import { TRENotification } from '../../../models/treNotification';
-import { awaitingStates, completedStates, failedStates, OperationStep } from '../../../models/operation';
+import { awaitingStates, completedStates, failedStates, inProgressStates, Operation, OperationStep } from '../../../models/operation';
 import { Link } from 'react-router-dom';
 import moment from 'moment';
 import { useInterval } from './useInterval';
+import { HttpMethod, useAuthApiCall } from '../../../hooks/useAuthApiCall';
+import { ApiEndpoint } from '../../../models/apiEndpoints';
+import { getResourceFromResult, Resource } from '../../../models/resource';
+import { NotificationPoller } from './NotificationPoller';
+import { OperationsContext } from '../../../contexts/OperationsContext';
 
 interface NotificationItemProps {
-  notification: TRENotification
+  operation: Operation,
+  showCallout: (o: Operation, r: Resource) => void
 }
 
 export const NotificationItem: React.FunctionComponent<NotificationItemProps> = (props: NotificationItemProps) => {
   const [now, setNow] = useState(moment.utc());
   const [isExpanded, setIsExpanded] = useState(false);
+  const [notification, setNotification] = useState({} as TRENotification);
+  const [loadingNotification, setLoadingNotification] = useState(true);
+  const [errorNotification, setErrorNotification] = useState(false);
+  const opsCtx = useContext(OperationsContext);
+
+  const apiCall = useAuthApiCall();
 
   const getRelativeTime = (createdWhen: number) => {
     return (moment.utc(moment.unix(createdWhen))).from(now);
   }
 
-  // update the 'now' time for comparison - only while the item is rendered (panel is open)
+  useEffect(() => {
+    const setupNotification = async (op: Operation) => {
+      // ignore if we've already set this operation up
+      if (notification.resource) return;
```

Comment:
stupid question: where is `notification` initialised?

Reply:
line 22, in the `useState` call

Reply:
line 22, in the `useState` setup